blocks/filter-details/filter-details.js

```diff
@@ -0,0 +1,75 @@
+import ffetch from '../../scripts/ffetch.js';
+import { readBlockConfig } from '../../scripts/lib-franklin.js';
+
+async function getFilterDetails() {
+  const urlSearchParams = new URLSearchParams(window.location.search);
+  const key = urlSearchParams.get('key');
+  const dataSource = urlSearchParams.get('dataSource');
+  const year = urlSearchParams.get('year');
+  const dataMapSheet = `/data-source/${dataSource}/data-mapping.json`;
+  const dataMap = await ffetch(dataMapSheet).sheet(year).all();
+  const spreadsheet = `/data-source/${dataSource}/${dataSource}-data.json`;
+  const data = await ffetch(spreadsheet)
+    .sheet(year)
+    .map((item) => item)
+    .filter((item) => item.Pkey === key);
+  const { value } = await data.next();
```

Comment:
There are a couple of things we might be able to do here. Some of them might not have a direct impact on performance, but here we go:

* Technically, we don't need the `.map()` call, because by returning the `item` itself it's not actually changing anything.
* We could replace `const { value } = await data.next()` by just adding `.first()` to the `ffetch` call above it.
* We're making two sequential requests due to the `await` on both `ffetch` calls. We might be able to get a _little_ bit of a performance gain by running those concurrently.

The updated code block could look something like this:

```
const promises = [];
promises.push(ffetch(dataMapSheet).sheet(year).all());
promises.push(ffetch(spreadsheet)
  .sheet(year)
  .filter((item) => item.Pkey === key)
  .first());

const data = await Promise.all(promises);
return {
  data: data[1],
  dataMap: data[0],
};
```